avocado/utils/genio.py

```diff
@@ -174,3 +174,22 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+def safe_write_file(filename, data):
+    """
+    Write data to a file.
```

Comment:
if the arg from test is as below ...
   try:
            memory.set_thp_value("invalid", "always")
        except genio.GenIOException as details:
            self.fail("failed  %s" % details)
or 
  try:
            memory.set_thp_value("enabled", "1")
        except genio.GenIOException as details:
            self.fail("failed  %s" % details)
In first case, the file is not present and in the second case the value written into it is invalid.
In both Cases I am not able to catch IOError in library.
library code:
class GenIOException(Exception):
    pass

    def write_file_or_log(self, filename, data):
        fd = os.open(filename, os.O_WRONLY)
        try:
            ret = os.write(fd, data)
        except IOError as details:
            raise GenIOException("The write to %s failed: %s" % (
                                 filename, details))
But when replaced IOError with OSError i can catch the exception in test and make it Fail.
So can we catch in the library using OSError???